crates/core/src/formatter/transaction.rs

```diff
@@ -0,0 +1,251 @@
+    payer: Address,
+    /// Execution context information
+    context: TransactionContext,
+    /// Gas consumption details
+    gas: GasDetails,
+    /// Changes in balances.
+    balance_diffs: Vec<BalanceDiff>,
+}
+
+impl TransactionSummary {
+    /// Creates a new transaction summary from execution results.
+    ///
+    /// # Arguments
+    ///
+    /// * `l2_gas_price` - The gas price on L2 in wei
+    /// * `tx` - The executed transaction
+    /// * `tx_result` - The execution results and logs
+    pub fn new(
+        l2_gas_price: u64,
+        tx: &Transaction,
+        tx_result: &VmExecutionResultAndLogs,
+        balance_diffs: Vec<BalanceDiff>,
+    ) -> Self {
+        let status: TransactionStatus = (&tx_result.result).into();
+        let tx_hash = tx.hash();
+        let initiator = tx.initiator_account();
+        let payer = tx.payer();
+
+        let used = tx.gas_limit() - tx_result.refunds.gas_refunded;
+        let limit = tx.gas_limit();
+        let refunded = tx_result.refunds.gas_refunded;
+
+        Self {
+            status,
+            tx_hash,
+            initiator,
+            payer,
+            context: TransactionContext { l2_gas_price },
+            gas: GasDetails {
+                limit,
+                used,
+                refunded,
+            },
+            balance_diffs,
+        }
+    }
+}
+
+impl Display for TransactionSummary {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let Self {
+            status,
+            tx_hash,
+            initiator,
+            payer,
+            context: TransactionContext { l2_gas_price },
+            gas,
+            balance_diffs,
+        } = self;
+
+        // Calculate gas costs in ETH
+        let paid = U256::from(*l2_gas_price) * gas.used;
+        let refunded = U256::from(*l2_gas_price) * gas.refunded;
+
+        // Format human-readable values
+        let gas_used = gas.used;
+        let gas_limit_human = to_human_size(gas.limit);
+        let gas_used_human = to_human_size(gas.used);
+        let gas_refunded_human = to_human_size(gas.refunded.into());
+        let emoji = self.status.emoji();
+        let l2_gas_price_human = format_gwei(self.context.l2_gas_price.into());
+
+        let mut balance_diffs_formatted_table = tabled::Table::new(
+            balance_diffs
+                .iter()
+                .map(Into::<internal::BalanceDiffRepr>::into)
+                .collect::<Vec<_>>(),
+        );
+        balance_diffs_formatted_table.with(tabled::settings::Style::modern());
+
+        // Basic transaction information
+        f.write_fmt(format_args!(
+            r#"
+{emoji} [{status}] Hash: {tx_hash:?}
+Initiator: {initiator:?}
+Payer: {payer:?}
+Gas Limit: {gas_limit_human} | Used: {gas_used_human} | Refunded: {gas_refunded_human}
+Paid: {paid_in_eth} ({gas_used} gas * {l2_gas_price_human})
+Refunded: {refunded_in_eth}
+{balance_diffs_formatted_table}
```

Comment:
Maybe add a header here before the table? Or, alternatively, an extra empty line between it and the rest of the summary would subjectively look better too